src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -315,12 +316,47 @@ final Type.Declared recordDeclaration(Cursor parent, Cursor recordCursor) {
         ClangSizeOf.with(structOrUnionDecl, recordCursor.type().size() * 8);
         ClangAlignOf.with(structOrUnionDecl, recordCursor.type().align() * 8);
         if (recordCursor.isAnonymousStruct()) {
-            AnonymousStruct.with(structOrUnionDecl);
+            AnonymousStruct.with(structOrUnionDecl, offsetOfAnonymousRecord(parent, recordCursor, recordCursor));
         }
 
         return Type.declared(structOrUnionDecl);
     }
 
+    /*
+     * For the first named field that is nested somewhere inside anonRecord, we get the offset
+     * to outermostParent and anonRecord itself. Subtracting the latter from the former
+     * then gives us the offset of the anonRecord within outermostParent.
+     *
+     * Deals with cases like this too:
+     *
+     * struct Foo {
+     *     char c; // offset = 0
+     *     struct <anon1> { // offset = 96 - 64 = 32
+     *         int: 32;
+     *         struct <anon2> { // offset = 96 - 32 = 64
+     *             int: 32;
+     *             int x; // offset(Foo) = 96, offset(anon2) = 32, offset(anon1) = 64
+     *         };
+     *     };
+     * };
+     */
+    private static OptionalLong offsetOfAnonymousRecord(Cursor outermostParent, Cursor anonRecord, Cursor record) {
+        AtomicReference<OptionalLong> result = new AtomicReference<>(OptionalLong.empty());
+        record.forEach(fc -> {
+            if (result.get().isPresent()) return;
```

Comment:
Why don't we return after setting a result, instead of waiting for the next cursor and then returning if we see a result has been set?

Reply:
We can not return from the outer function in the forEach. I don't think there is a way to short-circuit the forEach.

Reply:
We could enhance the forEach to take a `Predicate` instead, so that we can return true/false to short-circuit, and then have a Consumer overload that just always returns true. I'll look at that

Reply:
Done in: https://github.com/openjdk/jextract/pull/159/commits/876dd3612b915aced226cce665b8318382445f0e

Reply:
Sorry, got mixed up. You just want to avoid further processing if the result has already been set